### DISCUSSION TITLE
DAI 2023-2024 - My "first" IntelliJ + Maven project - @valentin-ricard

### DISCUSSION THREAD

**Discussion by knownasred** [heig-vd-dai-course/heig-vd-dai-course#30] [Show and tell]
Here's my *"first"* Intellij + Maven project!

https://github.com/valentin-ricard/hello-world-dai

<details>
  <summary>Fun fact (click to show)</summary>
  My real first java project was a small minecraft plugin, and I still have some mods out today :P
  https://github.com/evolzmc/herbocraft Is one way more recent example, made in one small week for a modjam ^^
  
</details>

**Comment by ludelafo** [comment]
Haha cool fun fact!

My first mods were on Worms 4: Mayhem in 2007-2008. For a very long time, I think it was the game I spent the most time on, ahead of all my other games combined. I always feel a bit "nostalgic" thinking about it.

Regarding your IntelliJ + Maven project, good points in the README.

The only thing I would add is how to build and run the application through the command line. Just in case someone doesn't use/don't want to use the same IDE, they can always run your app.

**Reply by ludelafo** [reply]
Oh also, I see that you added a transformer to the shade-plugin. Could you tell me a bit more about its usage?

Feel free to fill the poll in the GitHub Discussion regarding this chapter: https://github.com/orgs/heig-vd-dai-course/discussions/3.

You can do the same with the chapter Git, GitHub and Markdown: https://github.com/orgs/heig-vd-dai-course/discussions/2.

**Reply by knownasred** [reply]
o/
Your suggestion for the readme is an excellent idea, will do when I have some time ^^
For the transformer, it allows to remove warnings relating to the META_INF/*.mf duplication with shadowing, as we get conflicts for the manifest file.

**Reply by ludelafo** [reply]
Thank you very much! I'll update the course material to include this transformer as well.
